cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/WorkflowProgramController.java

```diff
@@ -82,7 +87,10 @@ public void running() {
       @Override
       public void terminated(Service.State from) {
         LOG.debug("Workflow service terminated from {}. Un-registering service {}.", from, serviceName);
-        cancelAnnounce.cancel();
+        if (cancelAnnounce != null) {
```

Comment:
Yes, I was failing a test for a reason that turned out to be unrelated to this. I will remove this.